packages/react-server-dom-relay/src/ReactFlightDOMRelayClientHostConfig.js

```diff
@@ -36,18 +36,21 @@ function parseModelRecursively(response: Response, parentObj, value) {
   }
   if (typeof value === 'object' && value !== null) {
     if (Array.isArray(value)) {
-      for (let i = 0; i < value.length; i++) {
-        (value: any)[i] = parseModelRecursively(response, value, value[i]);
+      const clonedValue = value.slice(0);
```

Comment:
You’re going to override every property below here so a full clone is unnecessary. A new array is sufficient.

Just loop over the original instead.